app/controllers/users_controller.rb

```diff
@@ -1,7 +1,7 @@
 class UsersController < ApplicationController
-  before_action :get_resources
-  before_action :ensure_user_not_logged_in, only: [:new, :create]
-  skip_before_action :authenticate_user, only: [:new, :create]
+  before_action :get_resources, except: [:confirm_email]
+  before_action :ensure_user_not_logged_in, only: [:new, :create, :confirm_email]
+  skip_before_action :authenticate_user, only: [:new, :create, :confirm_email]
```

Comment:
Task bardziej długoterminowy i do pomyślenia:

Teraz kiedy używasz Pundita i w nim wymagasz żeby użytkownik był zalogowany (lub nie - możesz przy niektórych akcjach używać `!@user`) to pomyśl w jakich sytuacjach potrzebujesz faktycznie używać `ensure_user_not_logged_in` oraz `authenticate_user`? Czy żeby utworzyć użytkownika musisz być wylogowany? Czy w czasie będzie to się zmieniało? (np. admin w przyszłości może móc tworzyć ręcznie użytkowników).

Oczywiście bardziej bym to rozważył koncepcyjnie i wyciągnął wnioski niż pędził na rewrite - chętnie podyskutuję o tych wnioskach w piątek, szczególnie że nie są one jednoznaczne i jestem przekonany że każda osoba zapytana w firmie udzieli innej odpowiedzi i uzasadnienia (jeśli masz ochotę to popytaj kilka osób i zobacz co powiedzą)